### PR TITLE
Refactor Admin

### PR DESCRIPTION
**Pull request by Friederi**
Check check!

### CODE REVIEW

**Comment by hSangg** [comment]
@Friederi please resolve conflict.
@ThiThanhCong please review this PR

File: tech-project-back-end/Controllers/AdminController.cs [hunk 1]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
-
-            return Ok(result);
+            try
+            {
+                var result = await _productService.GetTopSellerProducts(count);
+                return Ok(result);
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(new { message = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: Not returning error message for user

File: tech-project-back-end/Controllers/AdminController.cs [hunk 2]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
-            var result = new
+            catch (ArgumentException ex)
             {
-                labels,
-                revenues
-            };
-
-            return Ok(result);
+                _logger.LogError(ex, "Please recheck your argument");
+                return BadRequest(new { message = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: same above

File: tech-project-back-end/Controllers/AdminController.cs [hunk 3]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while fetching revenue by year");
+                return StatusCode(500, new { message = "An error occurred while processing your request.", details = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: same above

File: tech-project-back-end/Controllers/AdminController.cs [hunk 4]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
-            };
-
-            return Ok(result);
+                _logger.LogError(ex, "Please recheck your argument");
+                return BadRequest(new { message = ex.Message });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while fetching revenue by day in system");
+                return StatusCode(500, new { message = "An error occurred while processing your request.", details = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: same above

File: tech-project-back-end/Controllers/AdminController.cs [hunk 5]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
+            }
+            catch (ArgumentException ex)
+            {
+                _logger.LogError(ex, "Please recheck your argument");
+                return BadRequest(new { message = ex.Message });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while fetching total user in system");
+                return StatusCode(500, new { message = "An error occurred while processing your request.", details = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: same above

File: tech-project-back-end/Controllers/AdminController.cs [hunk 6]
```diff
@@ -9,133 +13,142 @@ namespace tech_project_back_end.Controllers
...
-            return Ok(total);
+            try
+            {
+                var result = await _userService.GetTotalUser();
+                return Ok(result);
+            }
+            catch (ArgumentException ex)
+            {
+                _logger.LogError(ex, "Please recheck your argument");
+                return BadRequest(new { message = ex.Message });
```
Comment (ThiThanhCong) [inline tech-project-back-end/Controllers/AdminController.cs]: same above

File: tech-project-back-end/Models/Product.cs [hunk 7]
```diff
@@ -8,7 +8,7 @@ public class Product
         [Key]
         [DatabaseGenerated(DatabaseGeneratedOption.None)]
         [Column("product_id", TypeName = "varchar(36)")]
-        public string product_id { get; set; }
+        public string Product_id { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/Product.cs]: rename variable in pascal case

File: tech-project-back-end/Models/Image.cs [hunk 8]
```diff
@@ -7,9 +7,9 @@ public class Image
     {
         [Key]
         [DatabaseGenerated(DatabaseGeneratedOption.None)]
-        public string ImageId { get; set; }
+        public string Image_Id { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/Image.cs]: rename variable in pascal case

File: tech-project-back-end/Models/Order.cs [hunk 9]
```diff
@@ -8,7 +8,7 @@ public class Order
         [Key]
         [DatabaseGenerated(DatabaseGeneratedOption.None)]
         [Column("order_id", TypeName = "varchar(36)")]
-        public string order_id { get; set; }
+        public string Order_id { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/Order.cs]: rename in pascal case

File: tech-project-back-end/Models/Product.cs [hunk 10]
```diff
@@ -8,7 +8,7 @@ public class Product
...
 
         [Column("name_pr", TypeName = "varchar(200)")]
         public string name_pr { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/Product.cs]: rename pascal case

File: tech-project-back-end/Models/Product.cs [hunk 11]
```diff
@@ -32,6 +32,9 @@ public class Product
         [ForeignKey("Supplier")] 
         public string supplier_id { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/Product.cs]: rename pascal case

File: tech-project-back-end/Models/DetailOrder.cs [hunk 12]
```diff
@@ -7,19 +7,19 @@ public class DetailOrder
...
+        public long Price { get; set; }
 
-        [Column("quality", TypeName = "int(11)")]
-        public int quality { get; set; }
+        [Column("quantity", TypeName = "int(11)")]
+        public int Quantity { get; set; }
 
-        [ForeignKey("order_id")]
+        [ForeignKey("Order_id")]
         public virtual Order Order { get; set; }
```
Comment (ThiThanhCong) [inline tech-project-back-end/Models/DetailOrder.cs]: rename all the variable in models into pascal case but keep the column name as (abc_xyz)

**Review by ThiThanhCong — APPROVED** [review]

**Pull request merged**